backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/AzureBackupStore.java

```diff
@@ -24,16 +31,58 @@
  * scheme: {@code basePath/partitionId/checkpointId/nodeId}.
  */
 public final class AzureBackupStore implements BackupStore {
-
-  private final AzureBackupConfig azureBackupConfig;
+  public static final String SNAPSHOT_FILESET_NAME = "snapshot";
+  public static final String SEGMENTS_FILESET_NAME = "segments";
+  private final ExecutorService executor;
+  private final FileSetManager fileSetManager;
+  private final ManifestManager manifestManager;
 
   public AzureBackupStore(final AzureBackupConfig config) {
-    azureBackupConfig = config;
+    this(config, buildClient(config));
+  }
+
+  public AzureBackupStore(final AzureBackupConfig config, final BlobServiceClient client) {
+    executor = Executors.newWorkStealingPool(4);
+    final BlobContainerClient blobContainerClient =
+        client.getBlobContainerClient(config.containerName());
```

Comment:
Yes, container name can be null, they use it to construct the URL of the "path" to the saved files.
The tests pass by running it without container name (if run one at a time because they then have the same path and try to save on existing files).

Reply:
Let's add a test for this and document it in the config templates :+1: